Linkta-org/core
# Hotfix & Improve Pre-commit hook and lint-staged configuration; Solve Bug #259

## Description

This PR finishes the implementation of pre-commit hook and lint-staged configuration. 
Improves DevEx with terminal feedback for pre-commit. 

The biggest change introduced is introducing separate lint-staged config files in both the client and server directories, mirroring a monorepo set up. This will give us more flexibility in changing things down the road. 

Fixes #259 


## Type of change
Bug Fix

## How Can this be tested? Testing Plan to review this PR

- go to random file in server and client add a TS error and change some formatting
- try to commit changes and watch pre-commit 🛑 you !
- add a comment somewhere and commit it to see the happy path if you want 


## Additional Information:
